eZ/Bundle/EzPublishCoreBundle/Tests/EventListener/ViewControllerListenerTest.php

```diff
@@ -151,6 +158,56 @@ public function testGetControllerMatchedView()
         $this->assertEquals($expectedView, $this->request->attributes->get('view'));
     }
 
+    public function testGetControllerEmitsProperEvents(): void
+    {
+        $viewObject = new class() extends BaseView {
+        };
+
+        $this->viewBuilderRegistry
+            ->expects($this->once())
+            ->method('getFromRegistry')
+            ->willReturn($this->viewBuilderMock);
+
+        $this->viewBuilderMock
+            ->expects($this->once())
+            ->method('buildView')
+            ->willReturn($viewObject);
+
+        $eventDispatcher = $this->createMock(EventDispatcherInterface::class);
+        $eventDispatcher
+            ->expect($this->exactly(2))
+            ->method('dispatch')
+            ->withConsecutive(
+                [
+                    $this->assertInstanceOf(ViewEvent::class),
+                    $this->assertSame(ViewEvents::FILTER_BUILDER_PARAMETERS),
+                ], [
+                    $this->assertInstanceOf(PostBuildViewEvent::class),
+                    $this->assertSame(PostBuildViewEvent::class),
+                ])
+            ->willReturnArgument(0);
+
+        $viewControllerListener = new ViewControllerListener(
+            $this->controllerResolver,
+            $this->viewBuilderRegistry,
+            $eventDispatcher,
+            $this->logger
+        );
+
+        $viewControllerListener->getController($this->event);
+
+        $eventsEmitted = array_map(static function (array $calledListener): string {
+            return $calledListener['event'];
+        }, $eventDispatcher->getCalledListeners());
+
+        self::assertSame([
+            ViewEvents::FILTER_BUILDER_PARAMETERS,
+            PostBuildViewEvent::class,
+        ], $eventsEmitted);
+
+        self::assertEmpty($eventDispatcher->getOrphanedEvents());
```

Comment:
The suggestion presented before required this part to be removed, since it's no longer a `TraceableEventDispatcher`.
```suggestion
```